File: sdk/python/examples/01_basic_transfer.py
"""
Example 1: Basic AIC Token Transfer

Demonstrates:
- Keypair generation
- Balance checking
- Simple transfer
- Transaction confirmation
"""

import asyncio
from aether import AetherClient, Keypair, TransactionHelper


async def main():
    async with AetherClient(rpc_url="http://localhost:8545") as client:
        sender = Keypair.from_seed("sender seed phrase")
        recipient = Keypair.generate()

        print(f"Sender: {sender.address}")
        print(f"Recipient: {recipient.address}")

        balance = await client.get_balance(sender.address)
        print(f"Sender balance: {balance} AIC")

        if balance < 1000:
            raise ValueError("Insufficient balance")

        nonce = await client.get_nonce(sender.address)

        tx = await TransactionHelper.create_transfer(
            sender, recipient.address, 1000, nonce
        )

        print("Sending transaction...")
        tx_hash = await client.send_transaction(tx)
        print(f"Transaction hash: {tx_hash}")

        receipt = await client.wait_for_transaction(tx_hash)
        print(f"Confirmed in slot: {receipt.block_slot}")
        print(f"Status: {receipt.status}")

        new_balance = await client.get_balance(recipient.address)
        print(f"Recipient new balance: {new_balance} AIC")


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/typescript/examples/01-basic-transfer.ts
/**
 * Example 1: Basic AIC Token Transfer
 * 
 * Demonstrates:
 * - Keypair generation
 * - Balance checking
 * - Simple transfer
 * - Transaction confirmation
 */

import { AetherClient, AetherKeypair, TransactionHelper } from '@aether/sdk';

async function main() {
  const client = new AetherClient({ rpcUrl: 'http://localhost:8545' });

  const sender = await AetherKeypair.fromSeed('sender seed phrase');
  const recipient = await AetherKeypair.generate();

  console.log('Sender:', sender.address);
  console.log('Recipient:', recipient.address);

  const balance = await client.getBalance(sender.address);
  console.log('Sender balance:', balance.toString(), 'AIC');

  if (balance < 1000n) {
    throw new Error('Insufficient balance');
  }

  const nonce = await client.getNonce(sender.address);

  const tx = await TransactionHelper.createTransfer(
    sender,
    recipient.address,
    1000n,
    nonce
  );

  console.log('Sending transaction...');
  const txHash = await client.sendTransaction(tx);
  console.log('Transaction hash:', txHash);

  const receipt = await client.waitForTransaction(txHash);
  console.log('Confirmed in slot:', receipt.blockSlot);
  console.log('Status:', receipt.status);

  const newBalance = await client.getBalance(recipient.address);
  console.log('Recipient new balance:', newBalance.toString(), 'AIC');
}

main().catch(console.error);

